packages/intl-locale/index.ts

```diff
@@ -639,41 +639,103 @@ export class Locale {
     return parseUnicodeLanguageId(locale).region
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getCalendars
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getCalendars
+   */
+  public getCalendars() {
+    return calendarsOfLocale(this)
+  }
+
   /**
    * https://tc39.es/proposal-intl-locale/#sec-Intl.Locale.prototype.calendars
    */
   public get calendars() {
     return calendarsOfLocale(this)
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getCollations
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getCollations
+   */
+  public getCollations() {
+    return collationsOfLocale(this)
+  }
+
   /**
    * https://tc39.es/proposal-intl-locale/#sec-Intl.Locale.prototype.collations
    */
   public get collations() {
     return collationsOfLocale(this)
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getHourCycles
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getHourCycles
+   */
+  public getHourCycles() {
+    return hourCyclesOfLocale(this)
+  }
+
   /**
    * https://tc39.es/proposal-intl-locale/#sec-Intl.Locale.prototype.hourCycles
    */
   public get hourCycles() {
     return hourCyclesOfLocale(this)
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getNumberingSystems
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getNumberingSystems
+   */
+  public getNumberingSystems() {
+    return numberingSystemsOfLocale(this)
+  }
+
   /**
    * https://tc39.es/proposal-intl-locale/#sec-Intl.Locale.prototype.numberingSystems
    */
   public get numberingSystems() {
     return numberingSystemsOfLocale(this)
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getTimeZones
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getTimeZones
+   */
+  public getTimeZones() {
+    return timeZonesOfLocale(this)
+  }
+
   /**
    * https://tc39.es/proposal-intl-locale/#sec-Intl.Locale.prototype.timeZones
    */
   public get timeZones() {
     return timeZonesOfLocale(this)
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getTextInfo
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getTextInfo
+   */
+  public getTextInfo() {
+    try {
+      const info = Object.create(Object.prototype)
+      const dir = characterDirectionOfLocale(this)
+
+      Object.defineProperty(info, 'direction', {
```

Comment:
please use `defineProperty` from `@formatjs/ecma402-abstract`. Same as below

Reply:
Thanks for the feedback 🙇 

I updated this `Object.defineProperty` and the ones in `getWeekInfo` too